auth/src/main/java/com/google/cloud/auth/samples/AuthExample.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.auth.samples;
+
+import com.google.api.gax.paging.Page;
+import com.google.auth.oauth2.ComputeEngineCredentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * Demonstrate various ways to authenticate requests using Cloud Storage as an example call.
+ */
+public class AuthExample {
+  // [START auth_cloud_implicit]
+  static void authImplicit() {
+    // If you don't specify credentials when constructing the client, the client library will
+    // look for credentials via the environment variable GOOGLE_APPLICATION_CREDENTIALS.
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    System.out.println("Buckets:");
+    Page<Bucket> buckets = storage.list();
+    for (Bucket bucket : buckets.iterateAll()) {
+      System.out.println(bucket.toString());
+    }
+  }
+  // [END auth_cloud_implicit]
+
+  // [START auth_cloud_explicit]
+  static void authExplicit(String jsonPath) throws IOException {
+    // You can specify a credential file by providing a path to GoogleCredentials.
```

Comment:
The explicit sample should *only* show the case where it reads from a file, it shouldn't show the implicit case as well.